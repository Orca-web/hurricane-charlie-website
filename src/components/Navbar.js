import React from 'react';

const Navbar = () => {
  return (
    <nav>
      <li>gallery</li>
      <li>contact</li>
      <li>shop</li>
    </nav>
  )
}

export default Navbar;