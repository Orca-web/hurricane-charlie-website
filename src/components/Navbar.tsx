import React from 'react';
import { NavLink } from 'react-router-dom';
import { NavData } from '../globals/navData';

const activeNav = { borderBottom: "3px solid #FF2E63" };

interface NavbarProps {
  navData: NavData[];
}

const Navbar = ({
  navData
}: NavbarProps) => (
    <nav>
      {navData.map(item => (
        <li key={item.id}>
          <NavLink to={item.url} exact activeStyle={activeNav}>
            {item.name}
          </NavLink>
        </li>
      ))}
    </nav>
);

export default Navbar;