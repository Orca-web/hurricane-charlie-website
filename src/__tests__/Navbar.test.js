import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../components/Navbar';

const app = shallow(<Navbar />);

describe("Navbar", () => {
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });
});