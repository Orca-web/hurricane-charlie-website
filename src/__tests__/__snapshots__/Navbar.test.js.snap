// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          gallery
        </li>,
        <li>
          contact
        </li>,
        <li>
          shop
        </li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "gallery",
        },
        "ref": null,
        "rendered": "gallery",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "contact",
        },
        "ref": null,
        "rendered": "contact",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "shop",
        },
        "ref": null,
        "rendered": "shop",
        "type": "li",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            gallery
          </li>,
          <li>
            contact
          </li>,
          <li>
            shop
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "gallery",
          },
          "ref": null,
          "rendered": "gallery",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "contact",
          },
          "ref": null,
          "rendered": "contact",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "shop",
          },
          "ref": null,
          "rendered": "shop",
          "type": "li",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
